src/ApiService/ApiService/onefuzzlib/orm/StorageProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Azure.Data.Tables;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.ResourceManager.Storage;
+using Azure.ResourceManager;
+using Azure.Identity;
+
+namespace Microsoft.OneFuzz.Service.OneFuzzLib.Orm;
+
+
+public interface IStorageProvider
+{
+    Task<TableClient> GetTableClient(string table);
+    IAsyncEnumerable<T> QueryAsync<T>(string filter) where T : EntityBase;
+    Task<bool> Replace<T>(T entity) where T : EntityBase;
+
+}
+
+public class StorageProvider : IStorageProvider
+{
+    private readonly string _accountId;
+    private readonly EntityConverter _entityConverter;
+
+    public StorageProvider(string accountId) {
+        _accountId = accountId;
+        _entityConverter = new EntityConverter();
+    }
+
+    public async Task<TableClient> GetTableClient(string table)
+    {
+        var (name, key) = GetStorageAccountNameAndKey(_accountId);
+        var tableClient = new TableServiceClient(new Uri(_accountId), new TableSharedKeyCredential(name, key));
+        await tableClient.CreateTableIfNotExistsAsync(table);
+        return tableClient.GetTableClient(table);
+    }
+
+
+    public (string?, string?) GetStorageAccountNameAndKey(string accountId) {
+        ArmClient armClient = new ArmClient(new DefaultAzureCredential());
```

Comment:
I'd argue for having one global arm client 

I found instantiation of ArmClient to be time consuming as well.

Reply:
moved to contructor